src/components/EnumSwitch.ts

```diff
@@ -0,0 +1,141 @@
+import { Component, ReactElement, createElement } from "react";
+import { Alert } from "./Alert";
+import { BootstrapStyle, EnumButton, SwitchStatus } from "./EnumButton";
+import { EnumSpan, EnumSpanProps } from "./EnumSpan";
+import * as classNames from "classnames";
+import "../ui/EnumSwitch.scss";
+
+export interface EnumSwitchProps {
+    alertMessage?: string;
+    enumList: { key: string, caption: string }[];
+    enumAttributeValue: string;
+    onClickAction: (caption?: string) => void;
+    status: SwitchStatus;
+    bootstrapStyle: BootstrapStyle;
+}
+
+export interface EnumSwitchState {
+    position?: number;
+    width?: number;
+    height?: number;
+    visibility?: "visible" | "hidden";
+}
+
+export class EnumSwitch extends Component<EnumSwitchProps, EnumSwitchState> {
+    private buttonNode: HTMLButtonElement;
+    private activeSpanNode: HTMLSpanElement;
+    private widgetContainerNode: HTMLDivElement;
+
+    constructor(props: EnumSwitchProps) {
+        super(props);
+
+        this.state = {
+            position: 0,
+            width: 0,
+            height: 0,
+            visibility: "visible"
+        };
+        this.enumToggleSlider = this.enumToggleSlider.bind(this);
+        this.createSpan = this.createSpan.bind(this);
+        this.getActiveSpanNodeRef = this.getActiveSpanNodeRef.bind(this);
+        this.WidgetContainerNodeRef = this.WidgetContainerNodeRef.bind(this);
+        this.getButtonNodeRef = this.getButtonNodeRef.bind(this);
+        this.registerEvents = this.registerEvents.bind(this);
+        this.removeEvents = this.removeEvents.bind(this);
+    }
+    render() {
+        if (this.props.status !== "noContext") {
+            return createElement("div", { className: "form-validation" },
+                createElement("div", {
+                    className: classNames("widget-enum-switch", "form-control",
+                        { disabled: this.props.status !== "enabled" }),
+                    ref: this.WidgetContainerNodeRef
+                }, this.createSpan()),
+                createElement(Alert, { message: this.props.alertMessage || "", bootstrapStyle: "danger" })
+            );
+        } else {
+            return createElement("span", { className: "enum-switch noContext" });
+        }
+    }
+
+    private createSpan(): ReactElement<{}>[] {
+        const btnElement: ReactElement<{}>[] = [];
+        if (this.props.enumAttributeValue) {
```

Comment:
When declaring constants. make sure that they spaced from the rest of the body of the code. So add space

Reply:
done